ygot/path_types.go

```diff
@@ -47,9 +54,38 @@ type NodePath struct {
 	p             PathStruct
 }
 
-// ResolvePath is a helper which returns the root PathStruct and absolute path
-// of a PathStruct node.
-func ResolvePath(n PathStruct) (PathStruct, []*gpb.PathElem, []error) {
+// FakeRootPathStruct is an interface that is implemented by the fake root path
+// struct type.
+type FakeRootPathStruct interface {
+	PathStruct
+	GetId() string
+	GetCustomData() map[string]interface{}
+}
+
+func NewDeviceRootBase(id string) *DeviceRootBase {
+	return &DeviceRootBase{NodePath: &NodePath{}, id: id, CustomData: map[string]interface{}{}}
+}
+
+// DeviceRootBase represents the fakeroot for all YANG schema elements.
+type DeviceRootBase struct {
+	*NodePath
+	id         string
+	CustomData map[string]interface{}
```

Comment:
if GetCustomData is exported, does CustomData need to be exported, too?

Reply:
- The generated package needs to embed the type in order to define new methods for it for path creation.
- The generated root type's `CustomData` needs to allow modification after construction.
- When embedding an out-of-package type, unexported fields are inaccessible to the importer package.

So, the only ways of doing this is to either to define a setter or make it exported.

I guess having a setter makes more sense, so I'll do that, and unexport `CustomData`.